src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandlerCommon.java

```diff
@@ -84,8 +95,59 @@ protected Mono<Tuple2<String, String>> vposAuthRequestPipeline(AuthorizationRequ
                 );
     }
 
-    protected Mono<Tuple3<String, String, Optional<String>>> npgAuthRequestPipeline(
+    protected Mono<Tuple4<String, String, Optional<String>,Optional<String>>> npgAuthRequestPipeline(
                                                                                     AuthorizationRequestData authorizationData
+    ) {
+        return Mono.just(authorizationData).flatMap( authData -> switch (authData.authDetails()){
+            case CardsAuthRequestDetailsDto cards ->
+                    invokeNpgConfirmPayment(authorizationData,cards
+                            .getOrderId(),false).map(confirmPaymentResponse -> Tuples.of(confirmPaymentResponse.getT1(),confirmPaymentResponse.getT2(),confirmPaymentResponse.getT3(),Optional.empty()));
+            case WalletAuthRequestDetailsDto ignored ->
+                    paymentGatewayClient.requestNpgBuildSession(authorizationData)
+                            .filter(orderIdAndFieldsDto -> Objects.equals(orderIdAndFieldsDto.getT2().getState(), WorkflowStateDto.READY_FOR_PAYMENT) && orderIdAndFieldsDto.getT2().getSessionId() != null && orderIdAndFieldsDto.getT2().getSecurityToken() != null)
```

Comment:
could you please move this response validation code inside payment gateway client?